admin/src/Components/CategoryPeriodsInput.js

```diff
@@ -9,7 +9,8 @@ const CategoryPeriodsInput = ({ categoryPeriods, showHistoric }) => {
 
     useEffect(() => {
         const unsubscribe = categoryPeriods.subscribe(() => {
-            setPeriods(categoryPeriods.periods);
+            // Must create a new array to force re-render
+            setPeriods(Array.from(categoryPeriods.periods));
```

Comment:
Is that the case because the array passed to the subscribe method itself is the same object? In that case, wouldn't it make sense to move this logic to the collection instead, so we don't have to handle this in each subscriber?

Reply:
Ah, I missed that part. Thanks for clarifying.